frontend/dockerfile/dockerfile2llb/convert_expose.go

```diff
@@ -52,29 +52,22 @@ func parsePortSpecs(ports []string) (exposedPorts []string, _ error) {
 }
 
 // splitProtoPort splits a port(range) and protocol, formatted as "<portnum>/[<proto>]"
-// "<startport-endport>/[<proto>]". It returns an empty string for both if
-// no port(range) is provided. If a port(range) is provided, but no protocol,
-// the default ("tcp") protocol is returned.
-//
-// splitProtoPort does not validate or normalize the returned values.
-func splitProtoPort(rawPort string) (proto string, port string) {
+// "<startport-endport>/[<proto>]". It returns an error if no port(range) or
+// an invalid proto is provided. If no protocol is provided, the default ("tcp")
+// protocol is returned.
+func splitProtoPort(rawPort string) (proto string, port string, _ error) {
 	port, proto, _ = strings.Cut(rawPort, "/")
 	if port == "" {
-		return "", ""
+		return "", "", errors.Errorf("no port specified: %s<empty>", rawPort)
 	}
-	if proto == "" {
-		proto = "tcp"
-	}
-	return proto, port
-}
-
-func validateProto(proto string) error {
+	proto = strings.ToLower(proto)
```

Comment:
It might be good to have a build check for this one as well to warn if proto is not lower case.